{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sample Report:\n",
    "\n",
    "#### McAuliffe, P. F. et al. \"Ability to generate patient-derived breast cancer xenografts is enhanced in chemoresistant disease and predicts poor patient outcomes\". PLoS ONE 10, e0136851 (2015)\n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4556673/pdf/pone.0136851.pdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Acquire Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### S2 Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S2 Table: Differentially expressed proteins between P0 and P1, cutoff: FDR=0.2</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Paired t tests</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P0 (Patients)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P1 (Mice)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>T statistics</td>\n",
       "      <td>P value</td>\n",
       "      <td>FDR</td>\n",
       "      <td>mean</td>\n",
       "      <td>sd</td>\n",
       "      <td>mean</td>\n",
       "      <td>sd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>-8.3362</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>-0.4167</td>\n",
       "      <td>0.7898</td>\n",
       "      <td>1.5519</td>\n",
       "      <td>0.7004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Merlin/NF2</td>\n",
       "      <td>5.4049</td>\n",
       "      <td>0.0057</td>\n",
       "      <td>0.0713</td>\n",
       "      <td>-0.2657</td>\n",
       "      <td>0.2003</td>\n",
       "      <td>-0.8749</td>\n",
       "      <td>0.228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  S2 Table: Differentially expressed proteins between P0 and P1, cutoff: FDR=0.2  \\\n",
       "0                                                NaN                               \n",
       "1                                                NaN                               \n",
       "2                                                NaN                               \n",
       "3                                               EGFR                               \n",
       "4                                         Merlin/NF2                               \n",
       "\n",
       "       Unnamed: 1 Unnamed: 2 Unnamed: 3     Unnamed: 4 Unnamed: 5 Unnamed: 6  \\\n",
       "0             NaN        NaN        NaN            NaN        NaN        NaN   \n",
       "1  Paired t tests        NaN        NaN  P0 (Patients)        NaN  P1 (Mice)   \n",
       "2    T statistics    P value        FDR           mean         sd       mean   \n",
       "3         -8.3362     0.0011     0.0662        -0.4167     0.7898     1.5519   \n",
       "4          5.4049     0.0057     0.0713        -0.2657     0.2003    -0.8749   \n",
       "\n",
       "  Unnamed: 7  \n",
       "0        NaN  \n",
       "1        NaN  \n",
       "2         sd  \n",
       "3     0.7004  \n",
       "4      0.228  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Differentially expressed proteins between patients(P0) and mice (P1)\n",
    "df = pd.read_excel(r'BCX_tables/S2_Table.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### S3 Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S3 Table. Copy number alterations detected in all tested passages of six models by targeted exome sequencing.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 30</th>\n",
       "      <th>Unnamed: 31</th>\n",
       "      <th>Unnamed: 32</th>\n",
       "      <th>Unnamed: 33</th>\n",
       "      <th>Unnamed: 34</th>\n",
       "      <th>Unnamed: 35</th>\n",
       "      <th>Unnamed: 36</th>\n",
       "      <th>Unnamed: 37</th>\n",
       "      <th>Unnamed: 38</th>\n",
       "      <th>Unnamed: 39</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCX-006</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P4</td>\n",
       "      <td>P7</td>\n",
       "      <td>P10</td>\n",
       "      <td>P13</td>\n",
       "      <td>P16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-010</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-022</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-024</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATM</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKAP3</td>\n",
       "      <td>3.2</td>\n",
       "      <td>9.7</td>\n",
       "      <td>8.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.3</td>\n",
       "      <td>1.1</td>\n",
       "      <td>1.1</td>\n",
       "      <td>0.9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CSMD1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATM</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>2.8</td>\n",
       "      <td>4</td>\n",
       "      <td>5.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATRX</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9</td>\n",
       "      <td>1.1</td>\n",
       "      <td>1.1</td>\n",
       "      <td>0.9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HEATR7B2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>1.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CPAMD8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  S3 Table. Copy number alterations detected in all tested passages of six models by targeted exome sequencing.  \\\n",
       "0                                                NaN                                                              \n",
       "1                                            BCX-006                                                              \n",
       "2                                               AKT1                                                              \n",
       "3                                                 AR                                                              \n",
       "4                                               ATRX                                                              \n",
       "\n",
       "  Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5 Unnamed: 6  \\\n",
       "0        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "1         P0         P1         P4         P7        P10        P13   \n",
       "2          4        NaN        NaN        NaN        NaN        NaN   \n",
       "3          1        1.3        1.1        1.1        0.9          1   \n",
       "4          1        0.9        1.1        1.1        0.9          1   \n",
       "\n",
       "  Unnamed: 7  Unnamed: 8 Unnamed: 9  ... Unnamed: 30 Unnamed: 31 Unnamed: 32  \\\n",
       "0        NaN         NaN        NaN  ...         NaN         NaN         NaN   \n",
       "1        P16         NaN    BCX-010  ...         NaN     BCX-022          P0   \n",
       "2        NaN         NaN        ATM  ...         NaN       AKAP3         3.2   \n",
       "3          1         NaN      CSMD1  ...         NaN         ATM         0.5   \n",
       "4          1         NaN   HEATR7B2  ...         NaN       BRCA2         1.2   \n",
       "\n",
       "  Unnamed: 33 Unnamed: 34 Unnamed: 35 Unnamed: 36  Unnamed: 37 Unnamed: 38  \\\n",
       "0         NaN         NaN         NaN         NaN          NaN         NaN   \n",
       "1          P1          P6         NaN     BCX-024           P0          P1   \n",
       "2         9.7         8.8         NaN        BAP1          1.5         0.7   \n",
       "3         0.5         0.9         NaN      CDKN2A          2.8           4   \n",
       "4           1         1.4         NaN      CPAMD8          NaN         3.1   \n",
       "\n",
       "  Unnamed: 39  \n",
       "0         NaN  \n",
       "1          P7  \n",
       "2         1.4  \n",
       "3         5.6  \n",
       "4           5  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Copy number alterations detected in all tested passages of six models by targeted exome sequencing (BCX-006, -010, -011, 017, -022, -024) \n",
    "# High amplification cutoff was 4 (highlighted in red) and high deletion cutoff was 1 (highlighted in green) in excel\n",
    "df = pd.read_excel(r'BCX_tables/S3_Table.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### S4 Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S4 Table. Copy number alterations detected in P0 and P1 of four models by whole exome sequencing.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>Unnamed: 11</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCX-010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>- Q151L</td>\n",
       "      <td>0.3561</td>\n",
       "      <td>0.4835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0.0639</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>- C29S</td>\n",
       "      <td>0.4615</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-04 00:00:00</td>\n",
       "      <td>0.2105</td>\n",
       "      <td>0.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADAM17 -</td>\n",
       "      <td>0.0505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMPH -</td>\n",
       "      <td>0.1136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>0.4485</td>\n",
       "      <td>NaN</td>\n",
       "      <td>- M110T</td>\n",
       "      <td>0.0627</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  S4 Table. Copy number alterations detected in P0 and P1 of four models by whole exome sequencing.  \\\n",
       "0                                                NaN                                                  \n",
       "1                                            BCX-010                                                  \n",
       "2                                               Gene                                                  \n",
       "3                                            - Q151L                                                  \n",
       "4                                           ADAM17 -                                                  \n",
       "\n",
       "  Unnamed: 1 Unnamed: 2  Unnamed: 3 Unnamed: 4 Unnamed: 5 Unnamed: 6  \\\n",
       "0        NaN        NaN         NaN        NaN        NaN        NaN   \n",
       "1        NaN        NaN         NaN    BCX-017        NaN        NaN   \n",
       "2         P0         P1         NaN       Gene         P0         P1   \n",
       "3     0.3561     0.4835         NaN          -     0.0639        NaN   \n",
       "4     0.0505        NaN         NaN     AMPH -     0.1136        NaN   \n",
       "\n",
       "   Unnamed: 7 Unnamed: 8 Unnamed: 9 Unnamed: 10  Unnamed: 11  \\\n",
       "0         NaN        NaN        NaN         NaN          NaN   \n",
       "1         NaN    BCX-022        NaN         NaN          NaN   \n",
       "2         NaN       Gene         P0          P1          NaN   \n",
       "3         NaN     - C29S     0.4615         NaN          NaN   \n",
       "4         NaN          -     0.3444      0.4485          NaN   \n",
       "\n",
       "           Unnamed: 12 Unnamed: 13 Unnamed: 14  \n",
       "0                  NaN         NaN         NaN  \n",
       "1              BCX-024         NaN         NaN  \n",
       "2                 Gene          P0          P1  \n",
       "3  2015-03-04 00:00:00      0.2105        0.35  \n",
       "4              - M110T      0.0627         NaN  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Copy number alterations detected in P0 and P1 of four models by whole exome sequencing (BCX-010, -017, -022, -024)\n",
    "# High amplification cutoff was 4 (highlighted in red) and high deletion cutoff was 1 (highlighted in green) in excel\n",
    "df = pd.read_excel(r'BCX_tables/S4_Table.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### S5 Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S5 Table. Somatic mutations detected in P0 and P1 of four models by whole exome sequencing.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 30</th>\n",
       "      <th>Unnamed: 31</th>\n",
       "      <th>Unnamed: 32</th>\n",
       "      <th>Unnamed: 33</th>\n",
       "      <th>Unnamed: 34</th>\n",
       "      <th>Unnamed: 35</th>\n",
       "      <th>Unnamed: 36</th>\n",
       "      <th>Unnamed: 37</th>\n",
       "      <th>Unnamed: 38</th>\n",
       "      <th>Unnamed: 39</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCX-006</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P4</td>\n",
       "      <td>P7</td>\n",
       "      <td>P10</td>\n",
       "      <td>P13</td>\n",
       "      <td>P16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-010</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-022</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-024</td>\n",
       "      <td>P0</td>\n",
       "      <td>P1</td>\n",
       "      <td>P7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CSMD2 R2096K</td>\n",
       "      <td>5.53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EGFR D1072E</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SMARCA4 R1465G</td>\n",
       "      <td>44.84</td>\n",
       "      <td>40.73</td>\n",
       "      <td>48.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERCC5 P903S</td>\n",
       "      <td>21.42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAV3 D414G</td>\n",
       "      <td>28.01</td>\n",
       "      <td>41.79</td>\n",
       "      <td>29.22</td>\n",
       "      <td>29.19</td>\n",
       "      <td>35.45</td>\n",
       "      <td>31.38</td>\n",
       "      <td>35.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGFR1 V273M</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53 E221*</td>\n",
       "      <td>96.71</td>\n",
       "      <td>70.12</td>\n",
       "      <td>98.43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FAT3 K2447N</td>\n",
       "      <td>5.59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CA H1047R</td>\n",
       "      <td>16.31</td>\n",
       "      <td>24.91</td>\n",
       "      <td>22.38</td>\n",
       "      <td>20.31</td>\n",
       "      <td>30.25</td>\n",
       "      <td>30.77</td>\n",
       "      <td>31.75</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MET V486A</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MLL3 S784Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.83</td>\n",
       "      <td>8.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGFR4 (splice donor variant)</td>\n",
       "      <td>29.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  S5 Table. Somatic mutations detected in P0 and P1 of four models by whole exome sequencing.  \\\n",
       "0                                                NaN                                            \n",
       "1                                            BCX-006                                            \n",
       "2                                       CSMD2 R2096K                                            \n",
       "3                                         NAV3 D414G                                            \n",
       "4                                      PIK3CA H1047R                                            \n",
       "\n",
       "  Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5 Unnamed: 6  \\\n",
       "0        NaN        NaN        NaN        NaN        NaN        NaN   \n",
       "1         P0         P1         P4         P7        P10        P13   \n",
       "2       5.53        NaN        NaN        NaN        NaN        NaN   \n",
       "3      28.01      41.79      29.22      29.19      35.45      31.38   \n",
       "4      16.31      24.91      22.38      20.31      30.25      30.77   \n",
       "\n",
       "  Unnamed: 7  Unnamed: 8   Unnamed: 9  ... Unnamed: 30     Unnamed: 31  \\\n",
       "0        NaN         NaN          NaN  ...         NaN             NaN   \n",
       "1        P16         NaN      BCX-010  ...         NaN         BCX-022   \n",
       "2        NaN         NaN  EGFR D1072E  ...         NaN  SMARCA4 R1465G   \n",
       "3      35.05         NaN  FGFR1 V273M  ...         NaN      TP53 E221*   \n",
       "4      31.75         NaN    MET V486A  ...         NaN      MLL3 S784Y   \n",
       "\n",
       "  Unnamed: 32 Unnamed: 33 Unnamed: 34 Unnamed: 35  \\\n",
       "0         NaN         NaN         NaN         NaN   \n",
       "1          P0          P1          P6         NaN   \n",
       "2       44.84       40.73        48.1         NaN   \n",
       "3       96.71       70.12       98.43         NaN   \n",
       "4         NaN        8.83        8.19         NaN   \n",
       "\n",
       "                    Unnamed: 36  Unnamed: 37 Unnamed: 38 Unnamed: 39  \n",
       "0                           NaN          NaN         NaN         NaN  \n",
       "1                       BCX-024           P0          P1          P7  \n",
       "2                   ERCC5 P903S        21.42         NaN         NaN  \n",
       "3                   FAT3 K2447N         5.59         NaN         NaN  \n",
       "4  FGFR4 (splice donor variant)        29.05         NaN         NaN  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Somatic mutations detected in P0 and P1 of four models by whole exome sequencing (BCX-010, -017, -022, -024)\n",
    "df = pd.read_excel(r'BCX_tables/S5_Table.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### S6 Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S6 Table. Somatic mutations detected in all tested passages of six models by targeted exome sequencing.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 17</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Unnamed: 19</th>\n",
       "      <th>Unnamed: 20</th>\n",
       "      <th>Unnamed: 21</th>\n",
       "      <th>Unnamed: 22</th>\n",
       "      <th>Unnamed: 23</th>\n",
       "      <th>Unnamed: 24</th>\n",
       "      <th>Unnamed: 25</th>\n",
       "      <th>Unnamed: 26</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCX-010-P0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-010-P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-017-P0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>BCX-022-P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-024-P0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCX-024-P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "      <td>...</td>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "      <td>gene</td>\n",
       "      <td>CNS</td>\n",
       "      <td>CN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2015-09-02 00:00:00</td>\n",
       "      <td>DEL</td>\n",
       "      <td>1.2</td>\n",
       "      <td>2015-09-02 00:00:00</td>\n",
       "      <td>H.DEL</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-03 00:00:00</td>\n",
       "      <td>DEL</td>\n",
       "      <td>1.5</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-03-01 00:00:00</td>\n",
       "      <td>H.DEL</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-05 00:00:00</td>\n",
       "      <td>AMP,DEL</td>\n",
       "      <td>3.5,1.4</td>\n",
       "      <td>2015-03-05 00:00:00</td>\n",
       "      <td>H.DEL</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2015-09-03 00:00:00</td>\n",
       "      <td>H.AMP,NORM</td>\n",
       "      <td>4,2.5</td>\n",
       "      <td>2015-09-03 00:00:00</td>\n",
       "      <td>H.AMP,AMP</td>\n",
       "      <td>4.8,3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-03-02 00:00:00</td>\n",
       "      <td>H.DEL</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5S_rRNA</td>\n",
       "      <td>DEL,NORM,AMP</td>\n",
       "      <td>1.5,2.1,3,2.2,2.9,2.3</td>\n",
       "      <td>5S_rRNA</td>\n",
       "      <td>H.DEL,AMP,NORM</td>\n",
       "      <td>0.8,2.8,1.9,2.9,2.1,3.4,2.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  S6 Table. Somatic mutations detected in all tested passages of six models by targeted exome sequencing.  \\\n",
       "0                                                NaN                                                        \n",
       "1                                         BCX-010-P0                                                        \n",
       "2                                               gene                                                        \n",
       "3                                2015-09-02 00:00:00                                                        \n",
       "4                                2015-09-03 00:00:00                                                        \n",
       "\n",
       "   Unnamed: 1 Unnamed: 2           Unnamed: 3 Unnamed: 4 Unnamed: 5  \\\n",
       "0         NaN        NaN                  NaN        NaN        NaN   \n",
       "1         NaN        NaN           BCX-010-P1        NaN        NaN   \n",
       "2         CNS         CN                 gene        CNS         CN   \n",
       "3         DEL        1.2  2015-09-02 00:00:00      H.DEL          1   \n",
       "4  H.AMP,NORM      4,2.5  2015-09-03 00:00:00  H.AMP,AMP      4.8,3   \n",
       "\n",
       "   Unnamed: 6           Unnamed: 7 Unnamed: 8 Unnamed: 9  ...  \\\n",
       "0         NaN                  NaN        NaN        NaN  ...   \n",
       "1         NaN           BCX-017-P0        NaN        NaN  ...   \n",
       "2         NaN                 gene        CNS         CN  ...   \n",
       "3         NaN  2015-03-03 00:00:00        DEL        1.5  ...   \n",
       "4         NaN                  NaN        NaN        NaN  ...   \n",
       "\n",
       "           Unnamed: 17 Unnamed: 18 Unnamed: 19  Unnamed: 20  \\\n",
       "0                  NaN         NaN         NaN          NaN   \n",
       "1           BCX-022-P1         NaN         NaN          NaN   \n",
       "2                 gene         CNS          CN          NaN   \n",
       "3  2015-03-01 00:00:00       H.DEL         0.7          NaN   \n",
       "4  2015-03-02 00:00:00       H.DEL         0.8          NaN   \n",
       "\n",
       "           Unnamed: 21   Unnamed: 22            Unnamed: 23  \\\n",
       "0                  NaN           NaN                    NaN   \n",
       "1           BCX-024-P0           NaN                    NaN   \n",
       "2                 gene           CNS                     CN   \n",
       "3  2015-03-05 00:00:00       AMP,DEL                3.5,1.4   \n",
       "4              5S_rRNA  DEL,NORM,AMP  1.5,2.1,3,2.2,2.9,2.3   \n",
       "\n",
       "           Unnamed: 24     Unnamed: 25                  Unnamed: 26  \n",
       "0                  NaN             NaN                          NaN  \n",
       "1           BCX-024-P1             NaN                          NaN  \n",
       "2                 gene             CNS                           CN  \n",
       "3  2015-03-05 00:00:00           H.DEL                          0.7  \n",
       "4              5S_rRNA  H.DEL,AMP,NORM  0.8,2.8,1.9,2.9,2.1,3.4,2.2  \n",
       "\n",
       "[5 rows x 27 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Somatic mutations detected in all tested passages of six models by targeted exome sequencing (BCX-006, -010, -011, 017, -022, -024)\n",
    "df = pd.read_excel(r'BCX_tables/S6_Table.xlsx')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Prepare Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter NaNs \n",
    "# Label \"Unnamed\" columns\n",
    "# Clean data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Explore Relationships"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Model (ML)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Conclusions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
